packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_token_burn_token.test.ts

```diff
@@ -0,0 +1,190 @@
+        }
+      }
+    }
+  })
+}
+
+describe('burnToken', () => {
+  beforeEach(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+    wavesConsortiumAddress = await testing.generateAddress()
+
+    await testing.token.dfi({ address: wavesConsortiumAddress, amount: 12 })
+    await testing.generate(1)
+    await testing.token.create({ symbol: 'BTC', collateralAddress: wavesConsortiumAddress })
+    await testing.generate(1)
+
+    await testing.token.mint({ symbol: 'BTC', amount: 100 })
+    await testing.generate(1)
+
+    idBTC = await testing.token.getTokenId(symbolDBTC)
+    providers = await getProviders(container)
+
+    await setupGovs()
+
+    // Fund 10 DFI UTXO
+    await fundEllipticPair(testing.container, providers.ellipticPair, 100)
+    await providers.setupMocks()
+
+    builder = new P2WPKHTransactionBuilder(providers.fee, providers.prevout, providers.elliptic, RegTest)
+  })
+
+  afterEach(async () => {
+    await testing.container.stop()
+  })
+
+  it('should reject if the amount is negative', async () => {
+    const script = await providers.elliptic.script()
+    const promise = builder.tokens.burnTokens({
+      amounts: [{ token: Number(idBTC), amount: new BigNumber(-2) }],
+      burnType: 0,
+      from: script,
+      variantContext: {
+        variant: 0,
+        context: {
+          stack: []
+        }
+      }
+    }, script)
+
+    await expect(promise).rejects.toThrow('The value of "value" is out of range. It must be >= 0 and <= 4294967295. Received -200000000')
+  })
+
+  it('should throw an error if not enough tokens are available to burn', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.tokens.burnTokens({
+      amounts: [{ token: Number(idBTC), amount: new BigNumber(15) }],
+      burnType: 0,
+      from: script,
+      variantContext: {
+        variant: 0,
+        context: {
+          stack: []
+        }
+      }
+    }, script)
+
+    // Ensure the created txn is correct
+    const promise = sendTransaction(testing.container, txn)
+
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow("DeFiDRpcError: 'BurnTokenTx: amount 0.00000000 is less than 15.00000000 (code 16)', code: -26")
+  })
+
+  it('should burnToken without context', async () => {
+    // Fund 100 BTC TOKEN
+    await testing.token.send({
+      address: await providers.getAddress(),
+      amount: 100,
+      symbol: 'BTC'
+    })
+    await testing.generate(1)
+
+    const script = await providers.elliptic.script()
+
+    const txn = await builder.tokens.burnTokens({
+      amounts: [{ token: Number(idBTC), amount: new BigNumber(15) }],
+      burnType: 0,
+      from: script,
+      variantContext: {
+        variant: 0,
+        context: script
+      }
+    }, script)
+
+    // Ensure the created txn is correct
+    await sendTransaction(testing.container, txn)
```

Comment:
```suggestion
    const outs = await sendTransaction(testing.container, txn)
```

Should test the `outs` just to be certain of what is expected.

Use `expect(outs).toStrictEqual([])` object assertion, instead of the old way of asserting line by line.